File: templates/todo/api/js/src/models/todoList.ts
import mongoose, { Schema } from "mongoose";

export type TodoList = {
    id: string
    name: string
    description?: string
    createdDate?: Date
    updatedDate?: Date
}

const schema = new Schema({
    name: {
        type: String,
        required: true,
    },
    description: String
}, {
    timestamps: {
        createdAt: "createdDate",
        updatedAt: "updatedDate"
    }
});

export const TodoListModel = mongoose.model<TodoList>("TodoList", schema, "TodoList");

File: templates/todo/api/csharp-cosmos-sql/TodoList.cs
namespace SimpleTodo.Api;

public class TodoList
{
    public TodoList(string name)
    {
        Name = name;
    }

    public string? Id { get; set; }
    public string Name { get; set; }
    public string? Description { get; set; }
    public DateTimeOffset CreatedDate { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? UpdatedDate { get; set; }
}